websocket/src/socket.cpp

```diff
@@ -0,0 +1,56 @@
+#include <dmsdk/dlib/socket.h>
+#include <dmsdk/dlib/sslsocket.h>
+#include "websocket.h"
+
+namespace dmWebsocket
+{
+
+dmSocket::Result WaitForSocket(WebsocketConnection* conn, dmSocket::SelectorKind kind, int timeout)
+{
+    dmSocket::Selector selector;
+    dmSocket::SelectorZero(&selector);
+    dmSocket::SelectorSet(&selector, kind, conn->m_Socket);
+    return dmSocket::Select(&selector, timeout);
+}
+
+dmSocket::Result Send(WebsocketConnection* conn, const char* buffer, int length, int* out_sent_bytes)
+{
+    int total_sent_bytes = 0;
+    int sent_bytes = 0;
+
+    while (total_sent_bytes < length) {
+        dmSocket::Result r;
+
+        if (conn->m_SSLSocket)
+            r = dmSSLSocket::Send(conn->m_SSLSocket, buffer + total_sent_bytes, length - total_sent_bytes, &sent_bytes);
+        else
+            r = dmSocket::Send(conn->m_Socket, buffer + total_sent_bytes, length - total_sent_bytes, &sent_bytes);
```

Comment:
I'm quite happy with how clean this code is now :)

Reply:
I agree. It looks great!